### PR TITLE
[CHK-2888] feat(npg-client): add tcp keep alive configurations

### PR DESCRIPTION
**Pull request by CianoDanilo**
#### List of Changes

Introduce tcp options (SO_KEEPALIVE, TCP_KEEPIDLE, TCP_KEEPINTVL and TCP_KEEPCNT) while creating HttpClient for NPG web client.
#### Motivation and Context

Reduce the overhead of connections creation by reusing connections
#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: helm/pay-wallet-values-prod.yaml [hunk 1]
```diff
@@ -94,9 +94,14 @@ microservice-chart:
     REDIS_PORT: "6380"
     REDIS_SSL_ENABLED: "true"
     WEB_LOGGING_LEVEL: "off"
+    DEFAULT_LOGGING_LEVEL: "debug"
```
Comment (pietro-tota) [inline helm/pay-wallet-values-prod.yaml:97]:
```suggestion
    DEFAULT_LOGGING_LEVEL: "INFO"
```

File: helm/pay-wallet-values-uat.yaml [hunk 2]
```diff
@@ -98,9 +98,14 @@ microservice-chart:
     REDIS_PORT: "6380"
     REDIS_SSL_ENABLED: "true"
     WEB_LOGGING_LEVEL: "off"
+    DEFAULT_LOGGING_LEVEL: "debug"
```
Comment (pietro-tota) [inline helm/pay-wallet-values-uat.yaml:101]:
```suggestion
    DEFAULT_LOGGING_LEVEL: "INFO"
```

**Review by pietro-tota — CHANGES_REQUESTED** [review]

File: helm/pay-wallet-values-dev.yaml [hunk 3]
```diff
@@ -77,9 +77,14 @@ microservice-chart:
     REDIS_PORT: "6380"
     REDIS_SSL_ENABLED: "true"
     WEB_LOGGING_LEVEL: "off"
+    DEFAULT_LOGGING_LEVEL: "debug"
```
Comment (giovanniberti) [inline helm/pay-wallet-values-dev.yaml:80]: Do we need `debug` by default?

**Review by giovanniberti — DISMISSED** [review]

**Comment by giovanniberti** [comment]
Remember to add the relevant env vars to https://github.com/pagopa/pagopa-ecommerce-local :)

File: helm/pay-wallet-values-prod.yaml (hunk 1 above)
Comment (CianoDanilo) [reply helm/pay-wallet-values-prod.yaml]: Changed with commit 997040d23000d61ba88ad0ed2db1b4baba7caad0

File: helm/pay-wallet-values-uat.yaml (hunk 2 above)
Comment (CianoDanilo) [reply helm/pay-wallet-values-uat.yaml]: Changed with commit 997040d23000d61ba88ad0ed2db1b4baba7caad0

File: helm/pay-wallet-values-uat.yaml [hunk 4]
```diff
@@ -98,9 +98,14 @@ microservice-chart:
...
     NPG_SERVICE_URI: "https://stg-ta.nexigroup.com/api/phoenix-0.0"
     NPG_SERVICE_READ_TIMEOUT: "10000"
     NPG_SERVICE_CONNECTION_TIMEOUT: "10000"
+    NPG_SERVICE_TCP_KEEP_ALIVE_ENABLED: "true"
+    NPG_SERVICE_TCP_KEEP_ALIVE_IDLE: "30"
```
[the remark this answers is not in the document]
Comment (CianoDanilo) [reply helm/pay-wallet-values-uat.yaml]: I thought uat had the same keep alive parameters as prod. Updated in 997040d23000d61ba88ad0ed2db1b4baba7caad0

File: helm/pay-wallet-values-dev.yaml (hunk 3 above)
Comment (CianoDanilo) [reply helm/pay-wallet-values-dev.yaml]: Changed in 997040d23000d61ba88ad0ed2db1b4baba7caad0

File: src/main/resources/logback-spring.xml [hunk 5]
```diff
@@ -1,14 +1,14 @@
...
-  </root>
+    <springProperty name="OTEL_SERVICE_VERSION" source="build.version"/>
+    <appender name="ECS_JSON_CONSOLE" class="ch.qos.logback.core.ConsoleAppender">
+        <encoder class="co.elastic.logging.logback.EcsEncoder">
+            <serviceName>${OTEL_SERVICE_NAME:-undefined}</serviceName>
+            <serviceVersion>${OTEL_SERVICE_VERSION}</serviceVersion>
+            <serviceEnvironment>${OTEL_SERVICE_ENVIRONMENT:-undefined}</serviceEnvironment>
+        </encoder>
+    </appender>
+    <root level="${DEFAULT_LOGGING_LEVEL}">
```
Comment (pietro-tota) [inline src/main/resources/logback-spring.xml:11]: what about refactoring this system property to root logging level? what do you think?

Comment (CianoDanilo) [reply src/main/resources/logback-spring.xml:11]: Working on it

Comment (CianoDanilo) [reply src/main/resources/logback-spring.xml]: Changed it 13e81b9c9fa90736e282d996278f51f9622591a1

**Review by pietro-tota — APPROVED** [review]

**Pull request merged**